internal/restic/blob_set.go

```diff
@@ -2,43 +2,80 @@ package restic
 
 import "sort"
 
-// BlobSet is a set of blobs.
-type BlobSet map[BlobHandle]struct{}
+// BlobSet defines all operation on a set of blobs.
```

Comment:
This interface is useless and unsafe: it causes runtime overhead that only serves to confuse the compiler. E.g.,

```
a := new(blobSet)
b := new(blobSet)
a.Merge(b)
```

compiles, but causes a type error at runtime.

Reply:
I see your point. First, I used `type BlobSet [NumBlobTypes]IDSet`. This however allows to use `len`and `range` for a `BlobSet` and gives very unexpected results. While `range` would also yield `IDSet`s instead of `BlobHandle`s and would be moste likely detected, most would not think that a `len(blobset)` does not give the number of blobs, but the number of BlobTypes that restic supports.

Do have a suggestion how to deal with this?

Reply:
```go
type BlobSet struct {
	byType [NumBlobTypes]IDSet
}
```

then use pointers everywhere. It's verbose, but it hides the implementation.

Reply:
Thanks, this is a good idea! I didn't use pointers, however. The struct should be already small enough..

Reply:
I guess so. It's just kind of strange that Insert modifies its caller even when it's passed by value. Also, if concurrency support is on the roadmap, you need to put the mutexes in a shared location...